Taarafo.Core/Services/Foundations/PostReports/PostReportService.Validations.cs

```diff
@@ -0,0 +1,108 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+// ---------------------------------------------------------------
+
+using System;
+using Taarafo.Core.Models.PostReports;
+using Taarafo.Core.Models.PostReports.Exceptions;
+
+namespace Taarafo.Core.Services.Foundations.PostReports
+{
+    public partial class PostReportService
+    {
+        private void ValidatePostReport(PostReport postReport)
+        {
+            ValidatePostReportNotNull(postReport);
+
+            Validate(
+                (Rule: IsInvalid(postReport.Id), Parameter: nameof(PostReport.Id)),
+                (Rule: IsInvalid(postReport.Details), Parameter: nameof(PostReport.Details)),
+                (Rule: IsInvalid(postReport.PostId), Parameter: nameof(PostReport.PostId)),
+                (Rule: IsInvalid(postReport.Post), Parameter: nameof(PostReport.Post)),
```

Comment:
Remove this.  The service is not responsible for validating models external to this service.  We only need to validate the Id's an not the navigation properties.